thorntail-runner/src/main/java/org/wildfly/swarm/runner/Runner.java

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright 2018 Red Hat, Inc, and individual contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wildfly.swarm.runner;
+
+import java.io.File;
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+import static java.util.Arrays.asList;
+
+/**
+ * The class to execute when running Thorntail in the IDE.
+ *
+ * It first generates an uber-jar and then executes it.
+ * The jar is generated from the IDE classpath by taking the Thorntail dependencies, generating an uber-jar in the similar way
+ * as thorntail-maven-plugin does and removing all the transitive dependencies of the Thorntail elements from WEB-INF/lib
+ * in the internal WAR.
+ *
+ * The following options are available to control the execution:
+ * <ul>
+ *     <li>
+ *         <b>thorntail.runner.preserve-jar</b> - keep the generated uber-jar when the application is stopped; might be useful for debugging
+ *     </li>
+ *     <li>
+ *         <b>thorntail.runner.user-dependencies</b> - a comma-separated list of groupId:artifactId:version of artifacts
+ *         to keep in WEB-INF/lib even though they are dependencies of Swarm elements
+ *     </li>
+ *     <li>
+ *         <b>thorntail.runner.main-class</b> - user's main class replacing the default org.wildfly.swarm.Swarm class.
+ *         Please note that using custom main class is discouraged.
+ *     </li>
+ *     <li>
+ *         <b>thorntail.runner.webapp-location</b> - by default, Runner expects webapp in `src/main/webapp`. This property
```

Comment:
Could this value be retrieved from Maven Project settings?

Reply:
`Runner` lives outside of a Maven invocation.

Reply:
As @Ladicek wrote, right now Thorntail Runner is build system agnostic and doesn't know anything about maven configuration. I'll add a TODO to try to take it from pom if present.